cadc-test-vos/src/main/java/ca/nrc/cadc/conformance/vos/SyncPullFromVOSpacePackageTest.java

```diff
@@ -0,0 +1,280 @@
+
+            // This test doesn't need data nodes created, it should only be manipulating the Job
+            URI testURI1 = new URI("vos://cadc.test/pkgTestURI1");
+            URI testURI2 = new URI("vos://cadc.test/pkgTestURI2");
+
+            List<URI> targets = new ArrayList<URI>();
+            targets.add(testURI1);
+            targets.add(testURI2);
+
+            // Create the Transfer.
+            Transfer transfer = new Transfer(Direction.pullFromVoSpace);
+            transfer.getTargets().addAll(targets);
+            // Forget adding protocols
+
+            // Add package view for tar file
+            View packageView = new View(new URI(Standards.PKG_10.toString()));
+            packageView.getParameters().add(new View.Parameter(new URI(VOS.PROPERTY_URI_ACCEPT), "tar"));
+            transfer.setView(packageView);
+
+            TransferResult result = doPkgTransfer(transfer);
+
+            // The job will have finished successfully, so Job phase should be PENDING.
+            assertEquals("Job phase should be PENDING", ExecutionPhase.PENDING, result.job.getExecutionPhase());
+
+            // Check the protocol objects to make sure NO endpoints are added
+            // If no protocols are provided this should happen
+            List<Protocol> protocolList = result.transfer.getProtocols();
+            Assert.assertTrue("protocol list should be empty: " + protocolList.size(), protocolList.isEmpty());
+
+            log.info("testPullFromVOSpacePkgNoProtocols passed.");
+        } catch (Exception unexpected) {
+            log.error("unexpected exception", unexpected);
+            Assert.fail("unexpected exception: " + unexpected);
+        }
+    }
+
+    @Test
+    public void testBadPullFromVOSpaceTarPkgNoTargets() {
+        try {
+            log.debug("testBadPullFromVOSpaceTarPkgNoTargets");
+
+            // Create the Transfer.
+            Transfer transfer = new Transfer(Direction.pullFromVoSpace);
+
+            // Forget to add the targets. Highly unlikely to happen, but testing anyway
+            // Add protocols
+            List<Protocol> protocols = new ArrayList<Protocol>();
+            protocols.add(new Protocol(VOS.PROTOCOL_HTTP_GET));
+            protocols.add(new Protocol(VOS.PROTOCOL_HTTPS_GET));
+            transfer.getProtocols().addAll(protocols);
+
+            // Add package view for tar file
+            View packageView = new View(new URI(Standards.PKG_10.toString()));
+            packageView.getParameters().add(new View.Parameter(new URI(VOS.PROPERTY_URI_ACCEPT), "tar"));
+            transfer.setView(packageView);
+
+            // Start the transfer.
+            StringWriter sw = getTransferXML(transfer);
+            log.debug("XML: " + sw.toString());
+
+            // POST the XML to the transfer endpoint.
+            WebResponse response = post(sw.toString());
+            assertEquals("POST response code should be 500", 500, response.getResponseCode());
```

Comment:
This should not result in a 500 internal error.  It must be a 400 level error (probably 400, bad request) because it was the user's fault.

Reply:
This is an error thrown from the TransferInlineContentHandler, at the point the transfer doc is being read in. Based on this, the decision is the test isn't valid and should be removed.